ShapeDocValues wrong serialization

### Description

Found by error-prone, looks rather serious to me since it impacts how the shape is serialized to disk. How come no tests fail?

```
/home/rmuir/workspace/lucene/lucene/core/src/java/org/apache/lucene/document/ShapeDocValues.java:553: warning: [LogicalAssignment] Assignment where a boolean expression was expected; use == if this assignment wasn't expected or add parentheses for clarity.
      if (node.triangle.ca = true) {
```

With more context:
```
      // write edge member of original shape
      if (node.triangle.ab == true) {
        header |= 0x10;
      }
      if (node.triangle.bc == true) {
        header |= 0x20;
      }
      if (node.triangle.ca = true) {                        // <-- bug
        header |= 0x40;
      }

      output.writeVInt(header);
```

After fixing this issue, we may enable error-prone `LogicalAssignment` check, since it is the only such issue across the entire codebase: https://errorprone.info/bugpattern/LogicalAssignment

### Version and environment details

_No response_

--- a/gradle/validation/error-prone.gradle
+++ b/gradle/validation/error-prone.gradle
@@ -375,7 +375,7 @@ allprojects { prj ->
             '-Xep:LiteEnumValueOf:WARN',
             '-Xep:LiteProtoToString:WARN',
             // '-Xep:LockNotBeforeTry:OFF',
-            // '-Xep:LogicalAssignment:OFF',
+            '-Xep:LogicalAssignment:WARN',
             // '-Xep:LongDoubleConversion:OFF',
             '-Xep:LongFloatConversion:WARN',
             '-Xep:LoopOverCharArray:WARN',

--- a/lucene/core/src/java/org/apache/lucene/document/ShapeDocValues.java
+++ b/lucene/core/src/java/org/apache/lucene/document/ShapeDocValues.java
@@ -550,7 +550,7 @@ abstract class ShapeDocValues {
       if (node.triangle.bc == true) {
         header |= 0x20;
       }
-      if (node.triangle.ca = true) {
+      if (node.triangle.ca == true) {
         header |= 0x40;
       }
 

